### PR TITLE
Sort competitors by registration if no payment has been made in V2

### PR DESCRIPTION
**Pull request by FinnIckler**

### CODE REVIEW

File: app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationList.jsx [hunk 1]
```diff
@@ -180,8 +180,21 @@ export default function RegistrationAdministrationList({ competitionInfo }) {
             return a.competing.registered_on.localeCompare(
               b.competing.registered_on,
             );
-          case 'paid_on':
-            return a.payment.updated_at.localeCompare(b.payment.updated_at);
+          case 'paid_on_with_registered_on_fallback':
+          {
+            if (a.payment && b.payment) {
+              return a.payment.updated_at.localeCompare(b.payment.updated_at);
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationList.jsx:186]: Not a fan of comparing dates as strings. If this is currently working as intended, then probably it's only a "happy accident".

You should parse these timestamps into either JS-native `Date` or Luxon `DateTime` representation and then use those to compare

File: app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationList.jsx [hunk 2]
```diff
@@ -180,8 +180,21 @@ export default function RegistrationAdministrationList({ competitionInfo }) {
...
+            }
+            if (a.payment && !b.payment) {
+              return 1;
+            }
+            if (!a.payment && b.payment) {
+              return -1;
+            }
+            return a.competing.registered_on.localeCompare(
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationList.jsx:194]: See above.

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/webpacker/components/RegistrationsV2/RegistrationAdministration/index.jsx [hunk 3]
```diff
@@ -4,15 +4,18 @@ import RegistrationAdministrationList from './RegistrationAdministrationList';
...
 export default function RegistrationEdit({ competitionInfo }) {
   const ref = useRef();
   return (
-    <QueryClientProvider client={new QueryClient()}>
+    <WCAQueryClientProvider>
       <StoreProvider reducer={messageReducer} initialState={{ message: null }}>
-        <RegistrationMessage parentRef={ref} />
+        <div ref={ref}>
+          <RegistrationMessage parentRef={ref} />
+        </div>
```
[the remark this answers is not in the document]
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/RegistrationAdministration/index.jsx:16]: it didn't, but it was throwing errors in the console making it harder to debug the issue. I could remove it from the diff and include it in another PR

[the remark this answers is not in the document]
Comment (gregorbg) [reply app/webpacker/components/RegistrationsV2/RegistrationAdministration/index.jsx:16]: Okay, nevermind then :D You can keep it in, my previous approval still holds

**Pull request merged**